src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/AsyncIteratorMethodBuilder.cs

```diff
@@ -3,31 +3,25 @@
 
 using System.Runtime.InteropServices;
 using System.Threading;
+using System.Threading.Tasks;
 
 namespace System.Runtime.CompilerServices
 {
     /// <summary>Represents a builder for asynchronous iterators.</summary>
     [StructLayout(LayoutKind.Auto)]
     public struct AsyncIteratorMethodBuilder
     {
-        // AsyncIteratorMethodBuilder is used by the language compiler as part of generating
-        // async iterators. For now, the implementation just wraps AsyncTaskMethodBuilder, as
-        // most of the logic is shared.  However, in the future this could be changed and
-        // optimized.  For example, we do need to allocate an object (once) to flow state like
-        // ExecutionContext, which AsyncTaskMethodBuilder handles, but it handles it by
-        // allocating a Task-derived object.  We could optimize this further by removing
-        // the Task from the hierarchy, but in doing so we'd also lose a variety of optimizations
-        // related to it, so we'd need to replicate all of those optimizations (e.g. storing
-        // that box object directly into a Task's continuation field).
-
-        private AsyncTaskMethodBuilder _methodBuilder; // mutable struct; do not make it readonly
+        /// <summary>The lazily-initialized box/task object, created the first time the iterator awaits something not yet completed.</summary>
+        /// <remarks>
+        /// This will be the async state machine box created for the compiler-generated class (not struct) state machine
+        /// object for the async enumerator.  Even though its not exposed as a Task property as on AsyncTaskMethodBuilder,
+        /// it needs to be stored if for no other reason than <see cref="Complete"/> needs to mark it completed in order to clean up.
+        /// </remarks>
+        private Task<VoidTaskResult>? m_task; // Debugger depends on the exact name of this field.
```

Comment:
>m_task [](start = 38, length = 6)

How the debugger depend on this name while you are introducing it here?

Reply:
It relies on this name from the method builders in general.